### ISSUE TITLE
[BUG] Module text right alignment is not respected

### ISSUE THREAD

**Issue by syntaxman** [bibletime/bibletime#402]
**Bug description**
Hebrew texts are left justified rather than right justified.  The text direction is correct, but the right side of the column is ragged rather than justified.

**Steps to reproduce**
Open a Hebrew text in it's own window or in parallel.  Text is left justified.

**Expected behavior**
Hebrew text should be right justified.

**System information:**
 - Arch Linux updated 8/4/22
 - BibleTime Version 3.0.3
 - Installed with pacman from community repository
 - Modules unknown to me

**Comment by jaakristioja** [comment]
As far as I can tell, the generated HTML has the correct `dir="rtl"` attributes. I think this is caused by the Qt rich text engine which does not do a good enough job supporting all of HTML and CSS 2.1, because the same HTML is correctly right-aligned in Firefox.

**Comment by gholmlund** [comment]
It works correctly using the "Simple" css display style. I am looking for the difference.

**Comment by syntaxman** [comment]
Just to clarify, the text direction is fine.  It's the justification that is wrong.  Instead of the lines being aligned so that they start at the right edge of the column, the right edge is ragged.

**Comment by gholmlund** [comment]
In html it is correct to use "dir" as in "dir=rtl". In css it is NOT correct. 
The correct use is "direction=rtl". 
I fixed the *.css and template files for stable-3.0 and master.

**Comment by jaakristioja** [comment]
> In html it is correct to use "dir" as in "dir=rtl". In css it is NOT correct. The correct use is "direction=rtl". I fixed the *.css and template files for stable-3.0 and master.

@gholmlund I think you are mistaken. I'm very convinced it should be `[dir=rtl]`. The switch to `[direction=rtl]` likely broke the HTML-exported documents.

Please instead check whether the QML sub-elements of [`ColumnItem`](https://github.com/bibletime/bibletime/blob/30f6a17c7ed2c5ac94b6baf64f7d709e5b11814b/src/frontend/display/modelview/ColumnItem.qml) are correctly anchored to their parent. By the looks of it the text elements are only anchored on the left side. This seems to cause them not to stretch to the full width of their parent, e.g.:

```
+-------- Parent --------+
|                        |
+-- Subelement --+       |
| Some RTL text. |       |
+----------------+       |
|                        |
+------------------------+
```

**Comment by gholmlund** [comment]
I was incorrectly thinking of the css property "direction" when I made this change. Effectively I removed the css entry for "rtl". I have reverted this change. 

The problem is not related to anchoring. A qml item horizontal size may be specified

**Comment by gholmlund** [comment]
Our problem with right aligned text is caused by a Qt text layout "feature".
 
Copied from Qt documentation:
_**If the widget's layout direction is Qt::RightToLeft (instead of Qt::LeftToRight, the default), Qt::AlignLeft refers to the right edge and Qt::AlignRight to the left edge. If you want Qt::AlignLeft to always mean "left" and Qt::AlignRight to always mean "right", combine the flag with Qt::AlignAbsolute.**_

I confirmed that when the css is changed to "left" the Hebrew text is right aligned.
    .entry[dir=rtl] {
        text-align: left !important;
    }
    
Here are a couple of Qt bug reports confirming that they see it as a feature, not a bug, and they don't plan to change it.

https://bugreports.qt.io/browse/QTBUG-47320
https://bugreports.qt.io/browse/QTBUG-93586

The AlignAbsolute flag mentioned above is documented for use with widgets. I have not been able to find a way to use it for qml items.

**Comment by gholmlund** [comment]
It seems the only way to fix the alignment for both displayed text and saved text is to substitute text for the text-align property.
    *.css
        .entry[dir=rtl] {
            text-align: #TEXTALIGN# !important;
        }

    *.cpp
        text.replace("#TEXTALIGN#", "left");  // or "right"

**Comment by gholmlund** [comment]
Searching for #CHAPTERTITLE" probable shows all the places to consider for replacement.

**Comment by gholmlund** [comment]
Fixed in stable-3.0 and master branches.
